arches/management/commands/packages.py

```diff
@@ -1114,6 +1124,128 @@ def import_reference_data(self, data_source, overwrite="ignore", stage="stage",
+                        self.import_sheet_to_model(wb[sheet], models.ControlledList)
+                    elif sheet == "ControlledListItem":
+                        self.import_sheet_to_model(wb[sheet], models.ControlledListItem)
+                    elif sheet == "ControlledListItemValue":
+                        self.import_sheet_to_model(wb[sheet], models.ControlledListItemValue)
+                # validate all data
+                for model in [models.ControlledList, models.ControlledListItem, models.ControlledListItemValue]:
+                    for instance in model.objects.all():
+                        instance.full_clean()
+                self.stdout.write('Data imported successfully from {0}'.format(source))
+        else:
+            self.stdout.write('The source file does not exist. Please rerun this command with a valid source file.')
+            sys.exit()
+    
+    def import_sheet_to_model(self, sheet, model):
+        fields = [{"name": field.name, "is_fk": field.get_internal_type() == "ForeignKey"} for field in model._meta.fields]
+        field_names = [field["name"] for field in fields]
+        
+        # Parse the sheet into a list of dictionaries
+        import_table = []
+        for imported_row in sheet.iter_rows(min_row=2, values_only=True):
+            working_row = {}
+            for field in field_names:
+                working_row[field] = imported_row[field_names.index(field)]
+            import_table.append(working_row)
+
+        # Process row data and create instances of the model
+        instances = []
+        list_items_with_parent = {}
+        for row in import_table:
+            instance = model()
+            for field in fields:
+                is_fk = field["is_fk"]
+                field_name = field["name"]
+                value = (row[field_name] if row[field_name] else None) # might be ''
+                if value and is_fk and model == models.ControlledListItem: 
+                    if field_name == "controlled_list":    
+                        related_list = models.ControlledList.objects.get(id=value)
+                        setattr(instance, field_name, related_list)
+                    elif field_name == "parent":
+                        # stash list items with parent relationships to create relationships after all list items have been created
+                        # stashed object in the form of {child_list_item_instance : parent_list_item_pk, ...}
+                        list_items_with_parent[instance] = value
+                elif value and is_fk and model == models.ControlledListItemValue:
+                    if field_name == "valuetype":
+                        valuetype = models.DValueType.objects.get(valuetype = value)
+                        setattr(instance, field_name, valuetype)
+                    elif field_name == "language":
+                        related_language = models.Language.objects.get(code=value)
```

Comment:
Any of these .get()'s could fail, but at least the error would be pretty self-explanatory. This one, though, I could see catching a user depending on whether or not they have concepts/languages in their new project. Do we want to add a hint around this with a follow-up action?

Reply:
I've added a check for missing languages - I'm not sure if we should automatically create a new language if it is missing from the system during this ETL. I seem to recall a conversation about `sys.exit()` being used after that check, but it seems like we're using that elsewhere in `packages.py` in similar situations.